TS_src/client/src/App.tsx

```diff
@@ -1,63 +1,99 @@
-import socket from "./main";
+import React, { Component, useState, useEffect } from 'react';
+import { BrowserRouter as Router, Routes, Route, Link } from 'react-router-dom';
+import Header from './components/Header';
+import StartPage from './components/pages/StartPage';
+import Login from './components/pages/Login';
+import Registration from './components/pages/Registration';
+import Dashboard from './components/pages/Dashboard';
+import ChatPage from './components/ChatPage';
+import { User } from './components/types';
+import socket from './main';
+import toast, { Toaster } from 'react-hot-toast';
+import Contacts from './components/Contacts';
+import MyTripCard from './components/MyTripCard';
+import TripImWatching from './components/TripImWatching';
 
 function App() {
-  const [userInfo, setUserInfo] = useState<User>({
-    name: "",
-    phone_number: "",
-    password: "",
-  });
-
-  return (
-    <>
-      <Router>
-        <Header />
-        <Routes>
-          <Route
-            path="/"
-            element={
-              <StartPage userInfo={userInfo} setUserInfo={setUserInfo} />
-            }
-          />
-          <Route
-            path="/login"
-            element={<Login userInfo={userInfo} setUserInfo={setUserInfo} />}
-          />
-          <Route
-            path="/register"
-            element={
-              <Registration userInfo={userInfo} setUserInfo={setUserInfo} />
-            }
-          />
-          <Route
-            path="/dashboard"
-            element={
-              <Dashboard userInfo={userInfo} setUserInfo={setUserInfo} />
-            }
-          />
-          <Route
-            path="/chat"
-            element={
-              <ChatPage
-                userInfo={userInfo}
-                setUserInfo={setUserInfo}
-                path="/chat"
-                socket={socket}
-              />
-            }
-          />
-        </Routes>
-      </Router>
-    </>
-  );
+	const [contacts, setContacts] = useState<Array<User>>([]);
+
+	// hook for conditionally rendering components
+	const [activeComponent, setActiveComponent] = useState('');
+
+	// toggle components in sidebar
+	const handleClick = (componentName: string) => {
+		setActiveComponent(componentName);
+	};
+
+	// hook for tracking userTrip data
+	interface Trip {
+		active: boolean;
+		start_timestamp: string;
+		start_lat: string;
+		start_lng: string;
+		sos_timestamp: string;
+		sos_lat: string;
+		sos_lng: string;
+	}
+
+	interface TripProps {
+		userTrip: Trip;
+		setUserTrip: React.Dispatch<React.SetStateAction<Trip>>;
+	}
+
+	const [userTrip, setUserTrip] = useState<Trip>({
+		active: true,
+		start_timestamp: '',
+		start_lat: '',
+		start_lng: '',
+		sos_timestamp: '',
+		sos_lat: '',
+		sos_lng: '',
+	});
+
+	const [userInfo, setUserInfo] = useState<User>({
+		name: '',
+		phone_number: '',
+		password: '',
+	});
+
+	const login = (userData: any) => {
+		console.log(userData);
```

Comment:
could we remove the console log if it's no longer needed? 

Would we be able to update the typing for userData?

Reply:
I can remove the console log userData in the next PR if that works.